src/components/Popover/Popover.stories.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import type { ComponentStory, ComponentMeta } from '@storybook/react';
+import { config } from 'storybook-addon-designs';
+
+import { StoryPage } from '@sb/StoryPage';
+
+import { Button } from '../Button';
+
+import { Popover, PopoverVariant } from './Popover';
+import { PopoverContent } from './PopoverContent';
+import { PopoverTrigger } from './PopoverTrigger';
+import { PopoverTriggerElement } from './PopoverTriggerElement';
+import { PopoverArrow } from './PopoverArrow';
+
+const figmaLink = ''; // TODO: Add figma link
+
+export default {
+  title: `Components/Popover`,
+  component: Popover,
+  parameters: {
+    design: config([
+      {
+        type: 'figma',
+        url: figmaLink,
+      },
+      {
+        type: 'link',
+        url: figmaLink,
+      },
+    ]),
+    docs: {
+      page: () => (
+        <StoryPage
+          description={`TODO: Add a description (supports markdown)`}
+        />
+      ),
+    },
+  },
+  args: {
+    //TODO: Add default args
+  },
+} as ComponentMeta<typeof Popover>;
+
+const Template: ComponentStory<typeof Popover> = (args) => {
+  return (
+    <div>
+      <Popover popoverVariant={args.popoverVariant}>
+        <PopoverTrigger asChild>
+          <PopoverTriggerElement>
+            <Button>Åpne</Button>
+          </PopoverTriggerElement>
+        </PopoverTrigger>
+        <PopoverContent>
+          Popoverinnhold
+          <PopoverArrow></PopoverArrow>
+        </PopoverContent>
+      </Popover>
```

Comment:
Should we perhaps simlify the API a bit? F.ex the PopoverArrow - can we make that a boolean prop instead, and render it conditionally if it is true?

And I'm not entirely sure of what the purpose of `PopoverTriggerElement` is - it just adds a ref (which is not set in the example here). So it doesnt seem to be needed, but perhaps I have missed something.

I would think something like this should work, and be a bit simpler API:

```jsx
      <Popover popoverVariant={args.popoverVariant} showArrow={true|false}>
        <PopoverTrigger asChild>
            <Button>Åpne</Button>
        </PopoverTrigger>
        <PopoverContent>
          Popoverinnhold
        </PopoverContent>
      </Popover>
```

WDYT?

Reply:
That was my first approach but when I tried it I got a React ref error and the Popover wouldn't open. I couldn't solve this in another way than to add the PopoverTriggerElement. I also tried not setting asChild to true, and with that aproach the Popoveractually opens but in that case Radix auto-creates a Radix Button that we don't want. I'm thinking that radix auto-creates the button may cause the ref error, but I'm not sure. 

I looked on some examples and saw that this [example](https://codesandbox.io/s/3sk4j?file=/src/components/Popover.tsx) had the behaviour we want and implemented that. Should we solve it another way?

Reply:
The example you linked to doesnt have an extra `PopoverTriggerElement`, and does not use `asChild` either. It seems to be what we want though, so makes sense to use that as a starting point. 

I'm not too familiar with radix-ui, but I can dig a bit deeper tomorrow if you need an extra pair of eyes 🙂

Reply:
It uses Radix Slot which is a radix component for creating an own asChild API, https://www.radix-ui.com/docs/primitives/utilities/slot

Regardless, a pair of extra eyes would be nice :)